lib/AST/Module.cpp

```diff
@@ -2685,6 +2685,38 @@ ModuleDecl::getDeclaringModuleAndBystander() {
   return *(declaringModuleAndBystander = {nullptr, Identifier()});
 }
 
+bool ModuleDecl::isClangOverlayOf(ModuleDecl *potentialUnderlying) {
+  return getUnderlyingModuleIfOverlay() == potentialUnderlying;
+}
+
+bool ModuleDecl::isSameModuleLookingThroughOverlays(
+  ModuleDecl *other) {
+  if (this == other) {
+    return true;
+  }
+
+  if (getUnderlyingModuleIfOverlay() == other) {
+    return true;
+  }
+
+  if (this == other->getUnderlyingModuleIfOverlay()) {
+    return true;
+  }
```

Comment:
Nit: Why not use your cute little helper method?

```suggestion
  if (this->isClangOverlayOf(other) || other->isClangOverlayOf(this)) {
    return true;
  }
```

Reply:
🤦 thanks for recognizing that!